src/libraries/System.ObjectModel/tests/TrimmingTests/TypeConverterAttributeCtorTest.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ComponentModel;
+using System.Globalization;
+
+namespace TypeConverterAttributeTest
+{
+    /// <summary>
+    /// Tests that the public constructors of types passed into System.ComponentModel.TypeConverterAttribute
+    /// are not trimmed out when needed in a trimmed application.
+    /// </summary>
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            // Type-based TypeConverterAttribute ctor overload, ensure public parameterized ctor of TypeConverter type is preserved.
+            TypeDescriptor.AddAttributes(typeof(DayOfWeek), new TypeConverterAttribute(typeof(MyDayOfWeekConverter)));
+            var attribute = new DefaultValueAttribute(typeof(DayOfWeek), "Friday");
+            if ((DayOfWeek)attribute.Value != DayOfWeek.Monday)
+            {
+                return -1;
```

Comment:
Would it make sense to separate these into two different tests in order to isolate the use of each one?